nc-game-front-end/src/components/SingleReview.jsx

```diff
@@ -4,21 +4,45 @@ import { fetchSingleReview } from "../api";
 import { useParams } from "react-router-dom";
 import CommentsByReview from "./CommentsByReview";
 import { format } from "date-fns";
+import { patchVotes } from "../api";
 
 const SingleReview = () => {
   const { review_id } = useParams();
 
-  const [singleReview, setSingleReview] = useState([]);
+  const [singleReview, setSingleReview] = useState({});
   const [isLoading, setIsLoading] = useState(true);
-
+  const [votes, setVotes] = useState(0);
+  const [hasVoted, setHasVoted] = useState(false);
+ 
   useEffect(() => {
     setIsLoading(true);
     fetchSingleReview(review_id).then((response) => {
       setSingleReview(response.review);
+      setVotes(response.review.votes);
       setIsLoading(false);
     });
   }, []);
 
+  const downVote = (review_id) => {
+    setVotes(votes - 1);
+    setHasVoted(true);
```

Comment:
What if the user changes their mind and after an upvote they want to take their vote back and even downvote once? This is optional functionality, so I leave it up to you to implement it or not.